Adopt more smart pointers in Cookie Store API related code

RupinMittal opened the pull request:
#### fffcabbac578cf202b46286d4101c2daef3eaa01
<pre>
Adopt more smart pointers in Cookie Store API related code
<a href="https://bugs.webkit.org/show_bug.cgi?id=281738">https://bugs.webkit.org/show_bug.cgi?id=281738</a>
<a href="https://rdar.apple.com/138174266">rdar://138174266</a>

Reviewed by Geoffrey Garen and Chris Dumez.

Smart pointer adoption as per the static analyzer.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::MainThreadBridge::ensureOnMainThread):
(WebCore::CookieStore::MainThreadBridge::getAll):
(WebCore::CookieStore::MainThreadBridge::set):
(WebCore::CookieStore::~CookieStore):
(WebCore::CookieStore::stop):
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::protectedThread const):
* Source/WebCore/workers/WorkerGlobalScope.h:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/WebCookieCache.cpp:
(WebKit::WebCookieCache::cookiesForDOM):
(WebKit::WebCookieCache::clear):
(WebKit::WebCookieCache::clearForHost):
* Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp:
(WebKit::shouldBlockCookies):
(WebKit::WebCookieJar::isEligibleForCache const):
(WebKit::shouldRelaxThirdPartyCookieBlocking):
(WebKit::WebCookieJar::cookies const):
(WebKit::WebCookieJar::setCookies):
(WebKit::WebCookieJar::cookiesEnabled):
(WebKit::WebCookieJar::remoteCookiesEnabledSync const):
(WebKit::WebCookieJar::remoteCookiesEnabled const):
(WebKit::WebCookieJar::cookieRequestHeaderFieldValue const):
(WebKit::WebCookieJar::getRawCookies const):
(WebKit::WebCookieJar::setRawCookie):
(WebKit::WebCookieJar::deleteCookie):
(WebKit::WebCookieJar::getCookiesAsync const):
(WebKit::WebCookieJar::setCookieAsync const):
(WebKit::WebCookieJar::addChangeListener):
(WebKit::WebCookieJar::removeChangeListener):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::protectedCookieJar):
* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: <a href="https://commits.webkit.org/285466@main">https://commits.webkit.org/285466@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d42d1fc98e6dfded1cbc9d230d96025a7162bcb1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72682 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52107 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25480 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76872 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23915 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74797 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23716 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57172 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15658 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75749 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47129 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62565 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37589 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43779 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20031 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22245 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65631 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20390 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78544 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16928 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19520 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65617 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16976 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62573 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64891 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16054 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13192 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6837 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47905 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/2692 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48972 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50267 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48717 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -110,7 +110,7 @@ void CookieStore::MainThreadBridge::ensureOnMainThread(Function<void(ScriptExecu
     }
 
-    downcast<WorkerGlobalScope>(*context).thread().workerLoaderProxy()->postTaskToLoader(WTFMove(task));
+    Ref { downcast<WorkerGlobalScope>(*context).thread() }->workerLoaderProxy()->postTaskToLoader(WTFMove(task));
```
Let's add a `protectedThread()` to `WorkerGlobalScope`

cdumez commented on Source/WebCore/loader/CookieJar.cpp:
```diff
@@ -113,7 +113,7 @@ CookieRequestHeaderFieldProxy CookieJar::cookieRequestHeaderFieldProxy(const Doc
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
```
False positive?

RupinMittal replied:
Yeah, I also think this and the others are false positives. I did ask in our channel about this earlier. I'll remove these changes from this patch.

cdumez commented on Source/WebCore/loader/CookieJar.cpp:
```diff
@@ -123,7 +123,7 @@ void CookieJar::setCookies(Document& document, const URL& url, const String& coo
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
```
ditto.

cdumez commented on Source/WebCore/loader/CookieJar.cpp:
```diff
@@ -140,7 +140,7 @@ bool CookieJar::cookiesEnabled(Document& document)
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
```
ditto.

cdumez commented on Source/WebCore/loader/CookieJar.cpp:
```diff
@@ -170,7 +170,7 @@ String CookieJar::cookieRequestHeaderFieldValue(Document& document, const URL& u
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
```
ditto.

cdumez commented on Source/WebCore/loader/CookieJar.cpp:
```diff
@@ -183,7 +183,7 @@ bool CookieJar::getRawCookies(Document& document, const URL& url, Vector<Cookie>
     auto pageID = document.pageID();
     std::optional<FrameIdentifier> frameID;
-    if (auto* frame = document.frame())
+    if (RefPtr frame = document.frame())
```
ditto.

RupinMittal committed:
Adopt more smart pointers in Cookie Store API related code
https://bugs.webkit.org/show_bug.cgi?id=281738
rdar://138174266

Reviewed by Geoffrey Garen and Chris Dumez.

Smart pointer adoption as per the static analyzer.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::MainThreadBridge::ensureOnMainThread):
(WebCore::CookieStore::MainThreadBridge::getAll):
(WebCore::CookieStore::MainThreadBridge::set):
(WebCore::CookieStore::~CookieStore):
(WebCore::CookieStore::stop):
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::protectedThread const):
* Source/WebCore/workers/WorkerGlobalScope.h:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/WebPage/WebCookieCache.cpp:
(WebKit::WebCookieCache::cookiesForDOM):
(WebKit::WebCookieCache::clear):
(WebKit::WebCookieCache::clearForHost):
* Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp:
(WebKit::shouldBlockCookies):
(WebKit::WebCookieJar::isEligibleForCache const):
(WebKit::shouldRelaxThirdPartyCookieBlocking):
(WebKit::WebCookieJar::cookies const):
(WebKit::WebCookieJar::setCookies):
(WebKit::WebCookieJar::cookiesEnabled):
(WebKit::WebCookieJar::remoteCookiesEnabledSync const):
(WebKit::WebCookieJar::remoteCookiesEnabled const):
(WebKit::WebCookieJar::cookieRequestHeaderFieldValue const):
(WebKit::WebCookieJar::getRawCookies const):
(WebKit::WebCookieJar::setRawCookie):
(WebKit::WebCookieJar::deleteCookie):
(WebKit::WebCookieJar::getCookiesAsync const):
(WebKit::WebCookieJar::setCookieAsync const):
(WebKit::WebCookieJar::addChangeListener):
(WebKit::WebCookieJar::removeChangeListener):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::protectedCookieJar):
* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: https://commits.webkit.org/285466@main
